src/io/save_gals_hdf5.c

```diff
@@ -0,0 +1,1149 @@
+
+    char *tmp_units[NUM_OUTPUT_FIELDS] = {"Unitless", "Unitless", "Unitless", "Unitless", "Unitless",
+                                          "Unitless", "Unitless", "Unitless", "Unitless",
+                                          "Unitless", "Myr", "Mpc/h", "Mpc/h", "Mpc/h", "km/s", "km/s", "km/s",
+                                          "Unitless?Spinx", "Unitless?Spiny", "Unitless?Spinz", "Unitless", "1.0e10 Msun/h", "1.0e10 Msun/h",
+                                          "Rvir?", "km/s",
+                                          "km/s", "km/s", "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h",
+                                          "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h",
+                                          "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h", "Msun/yr", "Msun/yr", "Msun/yr",
+                                          "Msun/yr", "DiskScaleRadius?", "Cooling?", "Heating?", "1.0e10 Msun/h",
+                                          "Myr", "Myr", "Msun/yr", "1.0e10 Msun/yr", "infallVvir?", "infallVmax?"};
+
+    // These are the HDF5 datatypes for each field.
+    hsize_t tmp_dtype[NUM_OUTPUT_FIELDS] = {H5T_NATIVE_INT, H5T_NATIVE_INT, H5T_NATIVE_LLONG, H5T_NATIVE_LLONG, H5T_NATIVE_INT, 
+                                         H5T_NATIVE_INT, H5T_NATIVE_LLONG, H5T_NATIVE_INT, H5T_NATIVE_INT, 
+                                         H5T_NATIVE_INT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_INT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT};
+
+    for(int32_t i = 0; i < NUM_OUTPUT_FIELDS; i++) {
+        field_names[i] = tmp_names[i];
+        field_descriptions[i] = tmp_descriptions[i]; 
+        field_units[i] = tmp_units[i];
+        field_dtypes[i] = tmp_dtype[i];
+    }
+
+    return EXIT_SUCCESS;
+}
+
+
+#define TREE_MUL_FAC        (1000000000LL)
```

Comment:
These two lines should get absorbed into the dedicated `GalaxyIndex` assigning function